runner_manager/jobs/startup.py

```diff
@@ -17,19 +17,28 @@
 log = logging.getLogger(__name__)
 
 
-def create_runner_groups(settings: Settings, github: GitHub):
+def sync_runner_groups(
+    settings: Settings, github: GitHub
+):  # creation des runner groups
+    """Sync runner groups between the settings of the database and GitHub.
+
+    Args:
+        settings (Settings): Settings of the application
+        github (GitHub): GitHub client
+    """
+
     runner_groups_configs = settings.runner_groups
     existing_groups: List[RunnerGroup] = RunnerGroup.find().all()
-
     for runner_group_config in runner_groups_configs:
-        if runner_group_config.name not in [group.name for group in existing_groups]:
-            runner_group: RunnerGroup = RunnerGroup(**runner_group_config.dict())
+        if runner_group_config in [group.name for group in existing_groups]:
+            runner_group: RunnerGroup = RunnerGroup.find_from_base(runner_group_config)
+            runner_group.update(**runner_group_config.dict())
             runner_group.save(github=github)
+            existing_groups.remove(runner_group)
         else:
-            runner_group: RunnerGroup = RunnerGroup.find_from_base(runner_group_config)
+            runner_group: RunnerGroup = RunnerGroup(**runner_group_config.dict())
+            runner_group.save(github=github)
 
-            existing_groups.remove(runner_group)
-            runner_group.update(**runner_group_config.dict())
```

Comment:
Review this whole block here, I don't think we should see any diff considering our implementation.

Reply:
One example I can find is line 31: https://github.com/scality/runner-manager/pull/438/files#diff-2c00abe7ba70ca4a271cec7b4edac8ff8fe9475fd6a0956f556db85b8b569a31R31

I think the condition should be `if runner_group_config.name ...` instead of `if runner_group_config ...`